chimney/src/main/scala/io/scalaland/chimney/TransformerFSupport.scala

```diff
@@ -0,0 +1,97 @@
+package io.scalaland.chimney
+
+import scala.collection.compat._
+import scala.language.higherKinds
+
+trait TransformerFSupport[F[_]] {
+
+  def pure[A](value: A): F[A]
+
+  def combine[A, B](fa: F[A], fb: => F[B]): F[(A, B)]
+
+  def map[A, B](fa: F[A], f: A => B): F[B]
+
+  def traverse[M[X] <: Iterable[X], A, B](mfa: M[A], f: A => F[B])(
+      implicit fac: Factory[B, M[B]]
+  ): F[M[B]]
+}
+
+object TransformerFSupport {
+
+  implicit def TransformerFOptionSupport: TransformerFSupport[Option] =
+    new TransformerFSupport[Option] {
+
+      def pure[A](value: A): Option[A] = Option(value)
+
+      def combine[A, B](fa: Option[A], fb: => Option[B]): Option[(A, B)] = {
+        fa.flatMap(a => fb.map(b => (a, b)))
+      }
+
+      def map[A, B](fa: Option[A], f: A => B): Option[B] = {
+        fa.map(f)
+      }
+
+      def traverse[M[X] <: Iterable[X], A, B](mfa: M[A], f: A => Option[B])(
+          implicit fac: Factory[B, M[B]]
+      ): Option[M[B]] = {
+
+        val b = fac.newBuilder
+        val it = mfa.iterator
+        var wasNone = false
+        while (!wasNone && it.hasNext) {
+          f(it.next()) match {
+            case None     => wasNone = true
+            case Some(fb) => b += fb
+          }
+        }
+        if (wasNone) None else Some(b.result())
+      }
+    }
+
+  implicit def TransformerFEitherErrorAccumulatingSupport[E, C[X] <: IterableOnce[X]](
+      implicit ef: Factory[E, C[E]]
+  ): TransformerFSupport[Either[C[E], *]] = new TransformerFSupport[Either[C[E], *]] {
+
+    def pure[A](value: A): Either[C[E], A] = Right(value)
+
+    def combine[A, B](fa: Either[C[E], A], fb: => Either[C[E], B]): Either[C[E], (A, B)] = {
+      (fa, fb) match {
+        case (Right(ok1), Right(ok2)) =>
+          Right((ok1, ok2))
```

Comment:
Is it necessary to support each IterableOnce? Maybe use one type of collection with const concat time

Reply:
Well, in core library this is the widest type `C` that we can provide implementation for `Either[C[E], A]`. This is one instance only and no one stops you from providing instance for more specialized types. In cats integration module we can still define `TransformerFSupport[ValidatedNec]` without requiring `IterableOnce` for error type.

As in Scala there's no standard type class/bound for expressing collection with constant concat time, and I'm not keen to provide one in Chimney, I guess that's fine, isn't it?

Reply:
Maybe use own ADT instead of Either with UnrolledBuffer or another collection optimal for concatenation in internal and case error sequence to List or another immutable collection in public api?

Reply:
Currently I don't think that it's Chimney's duty to provide default collection types for optimal error concatenation. In many use cases standard `Vector` (or even `List`) will be just fine. For others, where performance does matter:
- it will either work ootb (as long as the collection type inherits IterableOnce) - hence I think `Either[UnrolledBuffer[String], *]` should just work now,
- or user will just need to provide few lines of glue code to make anything he needs working (such an instance combining Either and IterableOnce might be a good example how to implement it for your own type).

I agree that we should underline performance implications in the documentation for lifted transformers.

Reply:
Note about performance of picked collection type added to docs.